libs/vkd3d/command.c

```diff
@@ -3334,6 +3334,13 @@ static VkPipelineStageFlags vk_queue_shader_stages(VkQueueFlags vk_queue_flags)
                 VK_PIPELINE_STAGE_TESSELLATION_EVALUATION_SHADER_BIT |
                 VK_PIPELINE_STAGE_GEOMETRY_SHADER_BIT |
                 VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT;
+
+        if (device->device_info.mesh_shader_features.meshShader &&
```

Comment:
Given how NON_PIXEL_SHADER_RESOURCE always implies compute too, I don't imagine this will cause too many issues.

Reply:
My concern was that on AMD we'll have an implied semaphore for task shaders, but it seems that the RADV implementation doesn't do anything too crazy on the vkCmdPipelineBarrier call itself, just increment some counters.